packages/use-contractkit/src/connectors/connectors.ts

```diff
@@ -175,6 +175,7 @@ export class InjectedConnector implements Connector {
   }
 
   close(): void {
+    this.onNetworkChangeCallback = undefined;
```

Comment:
This will work to prevent any strangeness with this connector trying to update the UI when the ethereum chain change event fires.